examples/hotrod/pkg/pool/pool_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) 2020 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pool
+
+import (
+	"sync"
+	"sync/atomic"
+	"testing"
+)
+
+var n int32 = 0
+var wg sync.WaitGroup
+
+func testJob() {
+	atomic.AddInt32(&n, 1)
+	wg.Done()
+}
+
+func TestExecute(t *testing.T) {
+	tests := []struct {
+		name     string
+		pool     *Pool
+		job      func()
+		expected int32
+	}{
+		{
+			name:     "worker: 1, job: 1",
+			pool:     New(1),
+			job:      testJob,
+			expected: 1,
+		},
+		{
+			name:     "worker: 2, job: 2",
+			pool:     New(2),
+			job:      testJob,
+			expected: 2,
+		},
+		{
+			name:     "worker: 3, job: 3",
```

Comment:
what new conditions are being tested with counts 3 and 4 that were not already tested with 1 and 2?